### PR TITLE
gpu: Use HOST_CACHED memory for descriptor set and bda state

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
The memory used for storing descriptor set state can be very large, (32 bits for every descriptor, some apps have 1m+ descriptors). Changing this memory from HOST_COHERENT to HOST_CACHED and manually flushing it saves about 1/3rd of the time spent updating this memory.

The buffer device address state could also grow quite large, so do the same for it.

### CODE REVIEW

**Review by spencer-lunarg — APPROVED** [review]

File: layers/gpu_validation/gpu_validation.cpp [hunk 1]
```diff
@@ -1299,6 +1299,7 @@ void GpuAssisted::UpdateInstrumentationBuffer(gpuav_state::CommandBuffer *cb_nod
                 for (const auto &update : buffer_info.update_at_submit) {
                     SetBindingState(data, update.first, update.second);
                 }
+                vmaFlushAllocation(vmaAllocator, buffer_info.allocation, 0, VK_WHOLE_SIZE);
```
Comment (juan-lunarg) [inline layers/gpu_validation/gpu_validation.cpp:1302]: Return value isn't being used/checked.

Also comment add a comment explaining the flushing.

File: layers/gpu_validation/gpu_validation.cpp [hunk 2]
```diff
@@ -1379,7 +1380,7 @@ void GpuAssisted::PostCallRecordCmdBindDescriptorSets(VkCommandBuffer commandBuf
             buffer_info.size = words_needed * 4;
             buffer_info.usage = VK_BUFFER_USAGE_STORAGE_BUFFER_BIT;
             VmaAllocationCreateInfo alloc_info = {};
-            alloc_info.requiredFlags = VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT;
+            alloc_info.requiredFlags = VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_CACHED_BIT;
```
Comment (juan-lunarg) [inline layers/gpu_validation/gpu_validation.cpp:1383]: Comment

File: layers/gpu_validation/gpu_validation.cpp [hunk 3]
```diff
@@ -1497,6 +1498,7 @@ void GpuAssisted::PostCallRecordCmdBindDescriptorSets(VkCommandBuffer commandBuf
                     }
                 }
             }
+            vmaFlushAllocation(vmaAllocator, di_input_block.allocation, 0, VK_WHOLE_SIZE);
```
Comment (juan-lunarg) [inline layers/gpu_validation/gpu_validation.cpp:1501]: Return value isn't being used/checked.

Also comment add a comment explaining the flushing.

File: layers/gpu_validation/gpu_validation.cpp (hunk 1 above)
Comment (jeremyg-lunarg) [reply layers/gpu_validation/gpu_validation.cpp:1302]: I'm not sure what we should do in case of an error with this call. We'd still want to try to unmap.

Comment (jeremyg-lunarg) [reply layers/gpu_validation/gpu_validation.cpp:1302]: should I just explicitly cast away the return value and comment?

Comment (juan-lunarg) [reply layers/gpu_validation/gpu_validation.cpp:1302]: I think asserting the return value is equal to VK_SUCCESS is good enough IMO

**Review by ncesario-lunarg — APPROVED** [review]

**Review by juan-lunarg — APPROVED** [review]

**Review by Tony-LunarG — APPROVED** [review]
LGTM

**Comment by juan-lunarg** [comment]
How much does this improve performance btw? Like for doom eternal?

**Comment by jeremyg-lunarg** [comment]
> How much does this improve performance btw? Like for doom eternal?

For that trace it drops the total CPU time from 70s to 64s. So a little bit less than 10%.

**Pull request merged**